### PR TITLE
SEQNG-724: Extracted methods from GPIClient

### PR DESCRIPTION
**Pull request by sraaphorst**
Some methods currently implemented in `GPIClient` will be used by all clients of thhe `GIAPI`, such as the up-and-coming `GHOSTClient`, so this is a refactor to make a trait implementing those common methods.

### CODE REVIEW

File: modules/giapi/src/main/scala/giapi/client/GiapiClient.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  def genericApply(configuration: Configuration): F[CommandResult] =
+    giapi.command(
+      Command(
+        SequenceCommand.APPLY,
+        Activity.PRESET_START,
+        configuration
+      ), GiapiClient.DefaultCommandTimeout)
+
+    def observe[A: Show](dataLabel: A, expTime: FiniteDuration): F[CommandResult] =
```
Comment (cquiroz) [inline modules/giapi/src/main/scala/giapi/client/GiapiClient.scala]: This method is missagligned

Comment (sraaphorst) [reply modules/giapi/src/main/scala/giapi/client/GiapiClient.scala]: Oooops. Okay, pushing another commit. Sorry!

**Comment by cquiroz** [comment]
We should move all the standard methods `test`, `verify`, etc tothe same trait

**Comment by sraaphorst** [comment]
@cquiroz Excellent. I figured there would be more, but I wasn't sure which ones would be standard.
Tomorrow if you are  around, we can discuss.

**Comment by jluhrs** [comment]
In other words, everything under "General commands" in GPIClient.scala should be moved to GiapiClient.scala.

**Comment by sraaphorst** [comment]
@cquiroz and @jluhrs: I've moved all the common commands from `GPIClient` to `GiapiClient`.

**Pull request merged**